src/api/resources/collect_record.py

```diff
@@ -238,3 +230,40 @@ def missing(self, request, *args, **kwargs):
             missing_ids = [row[0] for row in cursor.fetchall()]
 
         return Response(dict(missing_ids=missing_ids))
+
+    @action(
+        detail=False, methods=["POST"], permission_classes=[ProjectDataAdminPermission]
+    )
+    def ingest(self, request, project_pk, *args, **kwargs):
+        supported_content_types = (
+            "application/csv",
+            "application/x-csv",
+            "text/csv",
+            "text/comma-separated-values",
+            "text/x-comma-separated-values",
+            "text/tab-separated-values",
+        )
+
+        protocol = request.data.get("protocol")
+        uploaded_file = request.FILES.get("file")
+        profile = request.user.profile
+        dryrun = truthy(request.data.get("dryrun"))
+
+        if protocol is None:
+            return Response("Missing protocol", status=400)
+
+        if uploaded_file is None:
+            return Response("Missing file", status=400)
+
+        content_type = uploaded_file.content_type
+        if content_type not in supported_content_types:
+            return Response("File type not supported", status=400)
+
+        decoded_file = uploaded_file.read().decode("utf-8").splitlines()
+        records, errors = ingest_fishbelt(
+            decoded_file, project_pk, profile.id, protocol, dryrun
+        )
+
+        if errors:
+            return Response(errors, status=400)
+        return Response(CollectRecordSerializer(records, many=True).data)
```

Comment:
Does `.data` mean validated data, or is there a `validated _data` attribute? Seems like that would be the point of sending back the data, right, rather than just whatever they uploaded?

Reply:
It's returning back the serialized CollectRecord instances